pkg/controllers/clusteroperator/clusteroperator_controller_test.go

```diff
@@ -68,62 +71,295 @@ var _ = Describe("ClusterOperator controller", func() {
+					ContainElement(And(HaveField("Type", Equal(configv1.OperatorProgressing)), HaveField("Status", Equal(configv1.ConditionFalse)))),
+					// This should be upgradeable=false due to CapiInstaller controller being degraded.
+					ContainElement(And(HaveField("Type", Equal(configv1.OperatorUpgradeable)), HaveField("Status", Equal(configv1.ConditionFalse)))),
+					// This should be degraded=true due to CapiInstaller controller being degraded.
+					ContainElement(And(
+						HaveField("Type", Equal(configv1.OperatorDegraded)),
+						HaveField("Status", Equal(configv1.ConditionTrue)),
+						HaveField("Reason", Equal(operatorstatus.ReasonControllersDegraded)),
+						HaveField("Message", Equal(fmt.Sprintf("The following controllers degraded conditions are not as expected: %s=True", operatorstatus.CapiInstallerControllerDegradedCondition))),
+					)),
+				)), "should match the expected ClusterOperator status conditions")
+		})
+
+		It("should aggregate the top level operator conditions with available=false when some controller conditions are not available", func() {
+			By("Setting up the controllers conditions, one not available")
+			startingCO := configv1resourcebuilder.ClusterOperator().WithName(controllers.ClusterOperatorName).Build()
+			patchBase := startingCO.DeepCopy()
+			startingCO.Status.Conditions = []configv1.ClusterOperatorStatusCondition{
+				{LastTransitionTime: metav1.Now(), Type: operatorstatus.CoreClusterControllerAvailableCondition, Status: configv1.ConditionTrue, Reason: operatorstatus.ReasonAsExpected},
+				{LastTransitionTime: metav1.Now(), Type: operatorstatus.CoreClusterControllerDegradedCondition, Status: configv1.ConditionFalse, Reason: operatorstatus.ReasonAsExpected},
+				{LastTransitionTime: metav1.Now(), Type: operatorstatus.MachineSyncControllerAvailableCondition, Status: configv1.ConditionTrue, Reason: operatorstatus.ReasonAsExpected},
+				{LastTransitionTime: metav1.Now(), Type: operatorstatus.MachineSyncControllerDegradedCondition, Status: configv1.ConditionFalse, Reason: operatorstatus.ReasonAsExpected},
+				{LastTransitionTime: metav1.Now(), Type: operatorstatus.KubeconfigControllerAvailableCondition, Status: configv1.ConditionTrue, Reason: operatorstatus.ReasonAsExpected},
+				{LastTransitionTime: metav1.Now(), Type: operatorstatus.KubeconfigControllerDegradedCondition, Status: configv1.ConditionFalse, Reason: operatorstatus.ReasonAsExpected},
+				{LastTransitionTime: metav1.Now(), Type: operatorstatus.CapiInstallerControllerAvailableCondition, Status: configv1.ConditionTrue, Reason: operatorstatus.ReasonAsExpected},
+				{LastTransitionTime: metav1.Now(), Type: operatorstatus.CapiInstallerControllerDegradedCondition, Status: configv1.ConditionFalse, Reason: operatorstatus.ReasonAsExpected},
+				{LastTransitionTime: metav1.Now(), Type: operatorstatus.SecretSyncControllerAvailableCondition, Status: configv1.ConditionFalse, Reason: "Error", Message: "persistent reconcile error"},
```

Comment:
Similar here, perhaps mark the line that's expected to trigger the desired test state